src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -348,16 +304,15 @@ def _align_date_range(cls, metrics_query: MetricsQuery) -> tuple[MetricsQuery, i
         # We use as a reference the interval supplied via the initial version of the query.
         interval = metrics_query.rollup.interval
         if interval:
-            modified_start, modified_end, intervals_number = to_intervals(
+            modified_start, modified_end, intervals_number = adjust_time_bounds_with_interval(
                 metrics_query.start,
                 metrics_query.end,
                 interval,
             )
-            if modified_start and modified_end:
```

Comment:
Using our internal impl allows us to not have to care about optional values since we know that time intervals are always there.